### ISSUE TITLE
Integration with Katib

### ISSUE THREAD

**Issue by amiryi365** [kube-HPC/hkube#291]
### **Integration with Kubeflow Katib**
[Katib](https://github.com/kubeflow/katib) is a scalable and flexible ML HyperParameter (HP) Tuning on Kubernetes. 
It's a component of [Kubeflow](https://www.kubeflow.org/) platform.
Katib creates multiple sets of model parameters out of a given parameters feasible space, and executes an ML model fit and evaluation job for each, then saves evaluation metrics results.
**Katib UI** displays all jobs and enables to compare evaluation metrics of different parameters sets.
Katib jobs are executed as **kubeflow** **TensorFlow** or **PyTorch** jobs. 
### Katib Concept
- **Study:** A single optimization run over a feasible space. Contains parameters feasible space configuration (as min/max values, step, list of values, etc.) and a **Trial** set.
- **Trial:** A list of parameter values, x, that will lead to a single evaluation of the model. A job is created for each Trial.
- **Suggestion:** An HP algorithm to construct a parameter set. Supported algorithms are: RandomSearch, GridSearch, Hyperband, Bayesian optimization.

**Katib UI** enables to define a **Study** (define model image, suggestion type, parameters feasible space list), run it (creating **Trials** by **Suggestion** and executing multiple workers, one per Trial) and display the results by graphs and tables.
### DL Images for Katib
Several examples are introduced, using existing kubeflow ML fit/eval images with specific public train/test datasets, such as: mxnet-mnist-example (DL image classification) and ames-housing (houses prices prediction).
I found no documentation on APIs: what is the format/schema of image input and output.
Looking at some examples:

- **Input** seems to have a CLI-like format: a "--param=value" list for specific algorithm parameters. The algorithm uses a standard arguments parser library to build the args list. 

- **Output** is much unclear: in some examples the result is just printed to log, and more complicated examples use libraries where the result is handled deep inside. This output issue should be studied better if we want to plant any user algorithm. It's probably a "metric-param=value" format lines in the image log.
### Integration Options
As a **kubeflow** component, integration with **Katib** is actually integration with **Kubeflow** (see related Yehiam's issue #276 ). As Yehiam mentioned, there are two options:

- **Hkube calls Katib:** Calling a **Katib** job from Hkube algorithm, using a new **Hkube Katib adapter** algorithm, which creates a **Study** by given parameters (or use a "stored" Study), compile and execute it as a kubeflow job, wait for results and return them.

- **Katib calls Hkube:** Calling Hkube algorithm/pipeline from Katib job - write a new hkube algorithm  for **Katib** which would execute the given Hkube pipeline/algorithm, using the **Hkube api-server**, track for results (using webhook) and return them in Katib style.

@maty21 @yehiyam
